pkg/network/go/binversion/elf.go

```diff
@@ -0,0 +1,45 @@
+package binversion
```

Comment:
nit: are all files in `binversion` a copy from the original package? this may require us to add a comment on every file (we can confirm with the agent-core our agent-platform team)

Reply:
They aren't direct copies (except for the LICENSE file), but they were directly edited from copies:
- https://github.com/golang/go/blob/release-branch.go1.17/src/cmd/go/internal/version/exe.go was edited to make `pkg/network/go/binversion/elf.go`
- https://github.com/golang/go/blob/release-branch.go1.17/src/cmd/go/internal/version/version.go was edited to make `pkg/network/go/binversion/parse.go`

Reply:
While I was testing Go 1.18, I noticed that the code I linked above (from Go 1.17) doesn't work with binaries compiled using 1.18 (I'm guessing because they changed the format of the version string?). So I replaced the code in `binversion` with [the new package added in Go 1.18](https://github.com/golang/go/blob/master/src/debug/buildinfo/buildinfo.go) that is basically the same thing as `binversion`, except I trimmed some of the excess code and added a function that takes in an already-parsed `debug/elf.(*File)` value.